### ISSUE TITLE
`duration.toISOString()` doing some floating point  arithmetic for milliseconds…

### ISSUE THREAD

**Issue by matthewlehner** [moment/moment#4175]
**Description of the Issue and Steps to Reproduce:**
It looks like `duration.toISOString()` is not handling milliseconds properly… You can reproduce this in the most recent version of Chrome in the console on momentjs.com

```javascript
duration = moment.duration("1:00:1.1");
duration.toISOString(); // => "PT1H1.099999999999909S"

duration.milliseconds(); // => 100
```

It looks like any duration over 1 minute with milliseconds is experiencing the problem…

```javascript
moment.duration("0:0:1.1").toISOString() // => "PT1.1S"
moment.duration("0:0:59.1").toISOString() // => "PT59.1S"
moment.duration("0:0:60.1").toISOString() // => "PT1M0.10000000000000142S"
```

**Environment:**
`new Date().toString()`: Fri Sep 15 2017 22:44:37 GMT-0700 (Pacific Daylight Time)
`new Date().toLocaleString()`: 9/15/2017, 10:44:37 PM
`new Date()).getTimezoneOffset()`: 420
`navigator.userAgent`: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3213.3 Safari/537.36
`moment.version`: 2.18.1

**Comment by ashsearle** [comment]
Thanks for the bug report.

This sounds like a duplicate of #4145.  The fix is ready and waiting to be merged in the next release (fix is in PR #4146)

**Comment by icambron** [comment]
Closing in favor of those.
